ch_pipeline/analysis/flagging.py

```diff
@@ -2121,6 +2124,9 @@ def process(self, timestream):
             ]
             tools.apply_gain(weight, weight_mask, out=weight, prod_map=products)
 
+        if "rebin_weight" in timestream.datasets:
+            timestream.rebin_weight[:] *= weight_mask
```

Comment:
I think here `weight_mask` is `(nfreq, ninputs, ntime)` if dealing with un-stacked visibilities, or `(nfreq, 1, ntime)` if dealing with stacked visibilities, so that it can broadcast against the `weight` dataset.  So we need to remove the input/singleton axis when multiplying by `rebin_weight`.